### PR TITLE
[ENH] Add `calibration_plot` for probabilistic forecasts

### LINKED ISSUE #5631: [ENH] Add calibration plot method
**Is your feature request related to a problem? Please describe.**
To assess probabilistic forecasts it would be nice to have a method to create calibration plots

### PR DESCRIPTION
**Pull request by benHeid**
#### Reference Issues/PRs
fixes #5631 

#### What does this implement/fix? Explain your changes.
This implements a method to create calibration plots and adds an example in the probabilistic forecasting notebook.

#### Did you add any tests for the change?

I would appreciate feedback on how testing of plot functionalities should look like

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, I've added myself and possibly others to the [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) file - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

File: sktime/utils/plotting.py [hunk 1]
```diff
@@ -504,3 +504,66 @@ def get_y(length, split):
...
+    x-axis: interval from 0 to 1
+    y-axis: interval from 0 to 1
+    plot elements: the calibration fo the forecast (blue) and the ideal
+        calibration (orange)
+
+    Parameters
+    ----------
+    y_true : pd.Series
+        The actual values of the forecast
+    y_pred : sktime splitter object, descendant of BaseSplitter
```
Comment (fkiraly) [inline sktime/utils/plotting.py:524]: this docstring element is incorrect, no?

File: sktime/utils/plotting.py [hunk 2]
```diff
@@ -504,3 +504,66 @@ def get_y(length, split):
...
+    visualise it.
+
+    x-axis: interval from 0 to 1
+    y-axis: interval from 0 to 1
+    plot elements: the calibration fo the forecast (blue) and the ideal
+        calibration (orange)
+
+    Parameters
+    ----------
+    y_true : pd.Series
```
Comment (fkiraly) [inline sktime/utils/plotting.py:522]: could also be single-series iterable, e.g., 1-column `pd.DataFrame` etc?

**Comment by benHeid** [comment]
# TODO
* [ ] Fix issue #5635 before merging, then rebase and rerun the notebook. Since due to #5635 the `plot_windows` in the notebook is creating two figures.

File: sktime/utils/plotting.py (hunk 1 above)
Comment (benHeid) [reply sktime/utils/plotting.py:524]: I fixed it

File: sktime/utils/plotting.py (hunk 2 above)
Comment (benHeid) [reply sktime/utils/plotting.py:522]: Yes also possible are single columned pd.DataFrames and np.arrays

**Review by fkiraly — CHANGES_REQUESTED** [review]
Looks good!

Quick question, the notebook diff is large, have you checked this is all benign?

**Review by fkiraly — APPROVED** [review]
As discussed, large diff is due to new content at the end of notebook

File: sktime/utils/plotting.py (hunk 2 above)
Comment (fkiraly) [reply sktime/utils/plotting.py]: so the docstrig should say so, no?

**Pull request merged**